tests/src/test/scala/org/apache/spark/sql/rapids/ParquetFilterSuite.scala

```diff
@@ -0,0 +1,262 @@
+/*
+ * Copyright (c) 2023, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.rapids
+
+import java.nio.charset.StandardCharsets
+
+import com.nvidia.spark.rapids.{GpuFilterExec, SparkQueryCompareTestSuite}
+
+import org.apache.spark.sql.{Column, DataFrame, SparkSession}
+import org.apache.spark.sql.execution.FilterExec
+import org.apache.spark.sql.functions.col
+import org.apache.spark.sql.internal.SQLConf
+
+class ParquetFilterSuite extends SparkQueryCompareTestSuite {
+
+  def stripSparkFilter(spark: SparkSession, df: DataFrame): DataFrame = {
+    val schema = df.schema
+    val withoutFilters = df.queryExecution.executedPlan.transform {
+      case FilterExec(_, child) => child
+      case GpuFilterExec(_, child) => child
+    }
+    spark.internalCreateDataFrame(withoutFilters.execute(), schema)
+  }
+
+  def withSQLConf(pairs: (String, String)*)(f: => Unit): Unit = {
+    pairs.foreach { case (k, v) =>
+      SQLConf.get.setConfString(k, v)
+    }
+    try f finally {
+      pairs.foreach { case (k, _) =>
+        SQLConf.get.unsetConf(k)
+      }
+    }
+  }
+
+  def withAllParquetReaders(code: => Unit): Unit = {
+    // test the row-based reader
+    withSQLConf(SQLConf.PARQUET_VECTORIZED_READER_ENABLED.key -> "false") {
+      withClue("Parquet-mr reader") {
+        code
+      }
+    }
+    // test the vectorized reader
+    withSQLConf(SQLConf.PARQUET_VECTORIZED_READER_ENABLED.key -> "true") {
+      withClue("Vectorized reader") {
+        code
+      }
+    }
+  }
+
+  def withAllDevicePair(func: (Boolean, Boolean) => Unit): Unit = {
+    for {
+      writeGpu <- Seq(true, false)
+      readGpu <- Seq(true, false)
+    } func(writeGpu, readGpu)
+  }
+
+  def testRangePartitioningPpd(spark: SparkSession, writeDf: DataFrame,
+      partCol: String, predicate: Column, length: Int)(
+      writeGpu: Boolean, readGpu: Boolean): Unit = {
+    withAllParquetReaders {
+      withTempPath { path =>
+        withSQLConf(
+          SQLConf.PARQUET_OUTPUT_TIMESTAMP_TYPE.key -> "TIMESTAMP_MICROS",
+          "spark.rapids.sql.test.enabled" -> "false",
```

Comment:
```scala
 val TEST_CONF = conf("spark.rapids.sql.test.enabled")
    .doc("Intended to be used by unit tests, if enabled all operations must run on the " +
      "GPU or an error happens.")
    .internal()
    .booleanConf
    .createWithDefault(false)
```
If enable GPU, we should set this as true. Or we may enconter this kind of case: 
The test passed, but it's running on CPU silently, and we expect it's running on GPU.

Changes would like:

```scala
if (writeGpu || readGpu) {
  spark.rapids.sql.test.enabled = true
}
```

Reply:
Actually I think it should be

```
"spark.rapdis.slq.test.enabled" -> (!writeGpu).toString
```

or else if we are reading on the GPU, but writing on the CPU we will run into a problem where the we get an error for having things not be on the GPU.

Reply:
Updated to
```scala
"spark.rapids.sql.test.enabled" -> writeGpu.toString
```
and
```scala
"spark.rapids.sql.test.enabled" -> readGpu.toString
```
to check all operations running on GPU during GPU read and write separately.

And removed binary test because we don't support binaryType right now.